cli/src/main/java/com/expediagroup/streamplatform/streamregistry/cli/command/delete/Formatter.java

```diff
@@ -0,0 +1,214 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.cli.command.delete;
+
+import static com.fasterxml.jackson.core.Version.unknownVersion;
+import static com.fasterxml.jackson.dataformat.yaml.YAMLGenerator.Feature.WRITE_DOC_START_MARKER;
+
+import java.io.IOException;
+import java.util.List;
+
+import lombok.SneakyThrows;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ConsumerBindingKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ConsumerKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.DomainKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.InfrastructureKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.Key;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ProducerBindingKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ProducerKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.SchemaKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.StreamBindingKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.StreamKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ZoneKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.specification.Specification;
+
+public class Formatter {
+  private static final ObjectMapper mapper = new ObjectMapper(new YAMLFactory().disable(WRITE_DOC_START_MARKER))
+      .registerModule(new SimpleModule("keys", unknownVersion(), List.of(
+          new DomainKeySerializer(),
+          new SchemaKeySerializer(),
+          new StreamKeySerializer(),
+          new ZoneKeySerializer(),
+          new InfrastructureKeySerializer(),
+          new ProducerKeySerializer(),
+          new ConsumerKeySerializer(),
+          new StreamBindingKeySerializer(),
+          new ProducerBindingKeySerializer(),
+          new ConsumerBindingKeySerializer()
+      )));
+
+  @SneakyThrows
+  public <S extends Specification> String format(Key<S> key) {
+    return mapper.writeValueAsString(key);
+  }
+
+  private static abstract class NestedKeySerializer<K> extends StdSerializer<K> {
+    private final String name;
+
+    protected NestedKeySerializer(Class<K> kClass, String name) {
+      super(kClass);
+      this.name = name;
+    }
+
+    @Override
+    public void serialize(K key, JsonGenerator g, SerializerProvider p) throws IOException {
```

Comment:
SerializerProvider p is not being used here, right?

```suggestion
    public void serialize(K key, JsonGenerator jsonGenerator) throws IOException {
```

Reply:
This method signature is from `StdSerializer`. The `void serialize(K key, JsonGenerator g)` signature it the one I've defined in `NestedKeySerializer`